Common/ResourceClasses/ObjectResource.cs

```diff
@@ -5,11 +5,12 @@ public class ObjectResource
 
     public readonly Assimp.Scene Model;
 
-    private static readonly Assimp.AssimpContext Importer = new();
+    protected static readonly Assimp.AssimpContext Importer = new();
 
-    protected ObjectResource(string modelPath)
+    // TODO this absolutely sucks
+    protected ObjectResource(string modelPath, bool texture = false)
```

Comment:
why not use modelResource?

Reply:
I think I didn't want to have  nullable `Texture` but screw it, it's only gotten worse because of that